packages/workflow/src/internals.ts

```diff
@@ -527,14 +539,133 @@ export class Activator implements ActivationHandler {
     );
   }
 
+  public doUpdate(activation: coresdk.workflow_activation.IDoUpdate): void {
+    const { id: updateId, protocolInstanceId, name, headers, runValidator } = activation;
+    if (!updateId) {
+      throw new TypeError('Missing activation update id');
+    }
+    if (!name) {
+      throw new TypeError('Missing activation update name');
+    }
+    if (!protocolInstanceId) {
```

Comment:
nit: You could have combine all of those checks into a single "Misformed activation" error. Being very detailed is useful for user provided objects (ie. errors are more likely, are user need to know what they need to fix), but here, a missing field would indicate a server/core level error.